native/core/src/common/bit.rs

```diff
@@ -131,6 +131,18 @@ pub fn read_num_bytes_u32(size: usize, src: &[u8]) -> u32 {
     trailing_bits(v as u64, size * 8) as u32
 }
 
+#[inline]
+pub fn read_u64(src: &[u8]) -> u64 {
+    let in_ptr = src.as_ptr() as *const u64;
+    unsafe { in_ptr.read_unaligned() }
```

Comment:
I'm curios why to read the raw pointer unaligned? its not guaranteed to be aligned?

Reply:
hmm not sure if it is always aligned...